packages/dataclass-editor/src/master/DataClassMasterContent.tsx

```diff
@@ -130,24 +128,22 @@ export const DataClassMasterContent = () => {
     resetAndSetRowSelection(table, newDataClass.fields, moveIds, row => row.name);
   };
 
-  const combineFields = useMutation({
-    mutationKey: genQueryKey('function', context),
-    mutationFn: async () => {
-      const selectedRows = table.getSelectedRowModel().rows;
-      const payload: CombinePayload = {
-        fieldNames: selectedRows.map(row => row.original.name)
-      };
-      return client.function({ actionId: 'combineFields', context, payload });
-    },
-    onSuccess: () => {
-      toast.info('Fields successfully combined');
-      queryClient.invalidateQueries({ queryKey: genQueryKey('data', context) });
+  const combineFields = useFunction(
+    'function/combineFields',
+    {
+      context: context,
```

Comment:
if the variable is called the same as the key, you con't have to specify it again
```suggestion
      context,
```